en/guides/typesafesessions.md

```diff
@@ -0,0 +1,154 @@
+
+The `MySession` type can then be included in the application's Codable route handlers. For example:
+
+```swift
+struct Book: Codable {
+    let title: String
+    let author: String
+}
+
+router.get("/cart") { (session: MySession, respondWith: ([Book]?, RequestError?) -> Void) -> Void in
+    respondWith(session.books, nil)
+}
+
+router.post("/cart") { (session: MySession, book: Book, respondWith: (Book?, RequestError) -> Void) -> Void in
+    session.books.append(book)
+    session.save()
+    respondWith(book, nil)
+}
+```
+
+Here the cart accepts a book item of type `Book`, appends the book to the list of books in the session, and responds with the book that was added, or a `RequestError.internalServerError` if the session could not be saved (such as a failure of the session store).
+
+Note that if you define `MySession` as a class, it must be marked `final`.
+
+If you define `MySession` as a struct, then in order to modify it within a handler, you will first need to assign it to a local variable:
+```swift
+router.post("/cart") { (session: MySessionStruct, book: Book, respondWith: (Book?, RequestError) -> Void) -> Void in
+    var session = session                // Required when mutating a Struct
+    session.books.append(book)
+    session.save()
+    respondWith(book, nil)
+}
+```
+
+## Saving a session
+
+It is necessary to save the session when updates have been made:
+
+```swift
+    session.save()
+```
+
+## Terminating a session 
+
+To explicitly terminate a session, removing it from the store, call:
+
+```swift
+    session.destroy()
+```
+
+## Handling store failure
+
+It is possible that the session store could become inaccessible, resulting in a failure to persist or remove sessions from the store. In such cases, an error will be logged for you. However, you can also take additional steps in the case of an error by supplying a callback which takes an `Error?`:
+```swift
+router.post("/cart") { (session: MySession, book: Book, respondWith: @escaping (Book?, RequestError) -> Void) -> Void in
+    session.books.append(book)
+    session.save() { error in
+        if let error = error {
+            respondWith(nil, .internalServerError)
+        } else {
+            respondWith(book, nil)
+        }
+    }
+}
+```
+
+## Automatic saving of sessions
+
+If you have declared `MySession` as a class, it is possible to implement an automatic saving of the session by defining a deinitializer on the class:
+```swift
+    deinit {
+        self.save()
+    }
+```
+Be aware that the session will then be saved after every request, regardless of whether it has been modified.
+
+Care should be taken if combining this with the automatic saving technique described above: you may want to introduce a flag on your type to track whether `destroy()` has been called, to avoid calling `save()` during deinitialization - otherwise, the session will be persisted back into the store again:
```

Comment:
Care should be taken if combining this with the automatic saving technique described above...
What is the 'this' you are referring to here?  Can't it be reworded to say something like "For this reason, care should be taken if you are using the automatic saving technique: ..."